### ISSUE TITLE
[mypy] Incorrect imports in stub files

### ISSUE THREAD

**Issue by alfaix** [pytorch/pytorch#63279]
## To Reproduce

1. Create a file `test.py` with the following content:
```python
import torch
t1 = torch.empty(())
t2 = torch.zeros((3,))
```
2. Run `mypy test.py`. It outputs:
```
 $ mypy test.py
test.py:2: error: Module has no attribute "empty"
test.py:3: error: Module has no attribute "zeros"
Found 2 errors in 1 file (checked 1 source file)
```
`pyright` produces similar output.

## Expected behavior
This is valid code, the type checker shouldn't complain.

## Environment
```
PyTorch version: 1.9.0+cu102
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.2 LTS (x86_64)
GCC version: (Ubuntu 11.1.0-1ubuntu1~20.04) 11.1.0
Clang version: 12.0.1-++20210802051739+fed41342a82f-1~exp1~20210802152501.132
CMake version: version 3.20.3
Libc version: glibc-2.29

Python version: 3.7.11 (default, Jul  3 2021, 17:58:19)  [GCC 9.3.0] (64-bit runtime)
Python platform: Linux-5.4.0-80-generic-x86_64-with-Ubuntu-20.04-focal
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.20.2
[pip3] torch==1.9.0
[pip3] torch-geometric==1.6.3
[pip3] torch-scatter==2.0.6
[pip3] torch-sparse==0.6.9
[pip3] torchvision==0.7.0
[pip3] mypy==0.910 # manually edited
[conda] Could not collect
```

## Additional context
Same issue is reproduced with a couple more types (`Module` being one of them).
I think the issue is that the `_C/__init__.pyi` has incorrect imports [right here](https://github.com/pytorch/pytorch/blob/master/torch/_C/__init__.pyi.in#L21).
It reads:
```python
from . import _nn as _nn
from . import _onnx as _onnx
from . import _VariableFunctions as _VariableFunctions
```
But it should be (I think):
```python
from ._nn import *
from ._onnx import *
from ._VariableFunctions import *
```
Doing this change resolves the issue. I can make a tiny PR if the above fix is correct

cc @ezyang @malfet @rgommers @xuzhao9 @gramster

**Comment by VitalyFedyunin** [comment]
Please submit PR to fix it.

**Comment by malfet** [comment]
Hmm, I can not reproduce the problem using PytTorch-1.9.0 on MacOS:
```
% /usr/bin/python3 -mtorch.utils.collect_env;  /usr/bin/python3 -mmypy mypy-test.py
Collecting environment information...
PyTorch version: 1.9.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.0 (arm64)
GCC version: Could not collect
Clang version: 12.0.5 (clang-1205.0.22.9)
CMake version: version 3.20.1
Libc version: N/A

Python version: 3.8 (64-bit runtime)
Python platform: macOS-12.0-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] mypy==0.910
[pip3] mypy-extensions==0.4.3
[pip3] torch==1.9.0
[conda] mypy_extensions           0.4.3            py39h2804cbe_3    conda-forge
[conda] numpy                     1.20.2           py39h1f3b974_0    conda-forge
[conda] torch                     1.10.0.dev20210804          pypi_0    pypi
[conda] torchaudio                0.9.0                    pypi_0    pypi
Success: no issues found in 1 source file
```

**Comment by rgommers** [comment]
There are tests for typing (under `test/typing`), it seems unlikely that basic calls like `torch.empty` are typed incorrectly. What is probably happening is that your PyTorch install went wrong somehow. Your output says Python version is `3.7.11`, so you compiled Python from source and PyTorch as well I assume? Can you run the test suite and check if it passes? Or at least `test/test_typing.py`?

**Comment by alfaix** [comment]
I used package installations for both: python was installed with `apt` and torch was installed with `pip`. It did seem unlikely that something this simple wouldn't work out-of-the-box, yes.

I found the issue after trying it in a fresh virtualenv, apparently `pyright` autogenerated (incorrect) stubs for `torch/__init__.py`, which imported from `_C`, but not from `_VariableFunctions` (and others in the list), and `mypy` picked them up. The stubs in `_C` are correct, they are just incorrectly used in the ones generated by `pyright`.

Apologies for the false alarm :)
